### ISSUE TITLE
How to run the model with only eeg data in edf file?

### ISSUE THREAD

**Issue by yiboliu94** [skjerns/AutoSleepScorer#7]
Hi,

It's really a good project(not flattering, it's really fantastic).
In your thesis, you compared the accuracy of different input channel selection such as eeg, eeg+emg, eeg+eog, and all.
But the coed itself requires all channels?
I wonder how can I run the AutoSleepScorer with only eeg channel?

Sincerely
Liu

**Comment by PaulMuadDib** [comment]
Hi Liu,
Did you try like this ?
- create a SleepData object with EEG only:
from sleepscorer import Scorer, SleepData
**file = SleepData('yourEEGfile.edf', start = xxx, stop = yyy,  channels={'EEG':'EEG Fpz-Cz'}, preload=False)**
- Create and run Scorer
scorer = Scorer([file], hypnograms=True, demo=True)
scorer.run()
tools.show_sample_hypnogram('sample-psg.groundtruth.csv', start=960, stop=1800)

**Comment by skjerns** [comment]
Thanks!

The (preliminary) pre-trained weights that I supplied need all three channels to work correctly. For usage of only one channel (EEG), you would need to re-train a network. It is not possible to leave one channel empty, as the neural network expects all three channels being present. This decision was made as all channels turned out to give the highest accuracy.

@PaulMuadDib With the example you provided, it will try to infere the EMG/EOG channel automatically from the edfx file (if I remember correctly). A single-channel mode is not yet implemented (also as the weights provided require three input channels)

**Comment by yiboliu94** [comment]
Hello,

Thank you for your reply.
I tried to edit the network but I cannot handle so many functions which are invoking each other.
Cloud you please provide a version which can work with only eeg channel? I would like to pay $300(usd). Our lab is focusing on the eeg study.

Best wishes


________________________________

**Comment by skjerns** [comment]
It is possible to just remove the channels not used. However, it is necessary to re-train the network. Although training one network to be able to use a variable subset of electrodes will be possible, I'd say it is easier to train separate networks for different channel combinations.
